lib/src/sql/edges.rs

```diff
@@ -26,6 +30,24 @@ impl fmt::Display for Edges {
 	}
 }
 
+impl Serialize for Edges {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		if is_internal_serialization() {
+			let mut val = serializer.serialize_struct(TOKEN, 3)?;
+			val.serialize_field("dir", &self.dir)?;
+			val.serialize_field("from", &self.from)?;
+			val.serialize_field("what", &self.what)?;
+			val.end()
+		} else {
+			let output = self.to_string();
+			serializer.serialize_some(&output)
```

Comment:
We should never get to serialising this type as a basic JSON/CBOR output, as we will always have called `compute` on this value by this time, and converted it to a primary type (Null, False, True, Number, Strand, Duration, Datetime, Uuid, Array, Object, Geometry). Therefore maybe a `serialize_none()` is better here?